File: rust/src/models/minimum_commitment.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::minimum_commitment_scope::MinimumCommitmentScope;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct MinimumCommitment {
    /// Decimal string in the plan currency, e.g. "100.00".
    pub amount: String,

    pub scope: MinimumCommitmentScope,
}

impl MinimumCommitment {
    pub fn new(amount: String, scope: MinimumCommitmentScope) -> Self {
        Self { amount, scope }
    }
}

File: typescript/src/models/minimumCommitment.ts
// this file is @generated
import {
  type MinimumCommitmentScope,
  MinimumCommitmentScopeSerializer,
} from "./minimumCommitmentScope";

export interface MinimumCommitment {
  /** Decimal string in the plan currency, e.g. "100.00". */
  amount: string;

  scope: MinimumCommitmentScope;
}

export const MinimumCommitmentSerializer = {
  _fromJsonObject(object: any): MinimumCommitment {
    return {
      amount: object["amount"],
      scope: MinimumCommitmentScopeSerializer._fromJsonObject(object["scope"]),
    };
  },

  _toJsonObject(self: MinimumCommitment): any {
    return {
      amount: self.amount,
      scope: MinimumCommitmentScopeSerializer._toJsonObject(self.scope),
    };
  },
};
